core/src/main/java/nl/nn/adapterframework/pipes/ForEachChildElementPipe.java

```diff
@@ -294,6 +299,14 @@ protected void createHandler(HandlerRecord result, IPipeLineSession session, Ite
 		ItemCallbackCallingHandler itemHandler = new ItemCallbackCallingHandler(callback);
 		result.inputHandler=itemHandler;
 		
+		if (getXmlDebugger()!=null && (StringUtils.isNotEmpty(getContainerElement()) || StringUtils.isNotEmpty(getTargetElement()) || getExtractElementsTp()!=null)) {
+			String containerElementString = StringUtils.isNotEmpty(getContainerElement()) ? "filter to containerElement '"+getContainerElement()+"'" : null;
+			String targetElementString = StringUtils.isNotEmpty(getTargetElement()) ? "filter to targetElement '"+getTargetElement()+"'" :null;
+			String xpathString = getExtractElementsTp()!=null ? "filter XPath '"+getElementXPathExpression()+"'": null;
+			String label = "XML after preprocessing: " + Misc.concatStrings(Misc.concatStrings(containerElementString, ", ", targetElementString),", ", xpathString);
+			result.inputHandler=getXmlDebugger().inspectXml(session, label, result.inputHandler);
```

Comment:
Poeh, kunnen we de concatStrings niet uitbereiden dat hij een String[] als 3e argument heeft oid, dit is lelijk zeg..